### ISSUE TITLE
deriveContextObjectType/deriveObjectType do not work with Option arguments

### ISSUE THREAD

**Issue by avitaln** [sangria-graphql/sangria#123]
Tested with sangria 0.6.1 + sangria-marshalling-api 0.2.0

Example:

```scala
import sangria.execution.Executor
import sangria.macros._
import sangria.macros.derive._
import sangria.schema._
import scala.concurrent.Await
import scala.concurrent.ExecutionContext.Implicits.global
import scala.concurrent.duration._

case class Category(id: String, name: String)

class CatalogContext {
  val queries: CatalogQueries = new CatalogQueries
}

class CatalogQueries {
  @GraphQLField def categories(ctx: Context[CatalogContext, Unit])(categoryId: Option[String]): Seq[Category] = {
    Seq(Category("cid1","cname1"),Category("cid2","cname2"))
  }
}

object GraphqlTest extends App {
  implicit val CategoryType = deriveObjectType[Unit, Category]()
  val CatalogQueriesType = deriveContextObjectType[CatalogContext, CatalogQueries, Unit](_.queries)
  val CatalogSchema = Schema(CatalogQueriesType)
  val document = graphql"""{ categories { id name } }"""
  val ctx = new CatalogContext()
  Await.result(Executor.execute(CatalogSchema, queryAst = document, userContext = ctx), 10.seconds)
}
```

**Comment by OlegIlyenko** [comment]
Thanks for the report! I can confirm and reproduce it. It's a bug in the macro which generates `ctx.arg[T]("name)` instead of `ctx.argOpt[T]("name)` for optional args. I added tests for all kinds of edge-cases with default arguments and optional values, but somehow completely missed this simple case :)

**Comment by OlegIlyenko** [comment]
I just published a snapshot build with the fix. Would appreciate if you could verify that everything works as expected in your project (just in case). Here is necessary SBT config:

```scala
resolvers += "Sonatype snapshots" at "https://oss.sonatype.org/content/repositories/snapshots/"
libraryDependencies += "org.sangria-graphql" %% "sangria" % "0.6.2-SNAPSHOT"
```

**Comment by avitaln** [comment]
thanks! works great

**Comment by OlegIlyenko** [comment]
FYI, just released this fix in [v0.6.2](https://github.com/sangria-graphql/sangria/releases/tag/v0.6.2)
